### PR TITLE
Post purchase upsell: DIFM offer

### PR DESCRIPTION
**Pull request by niranjan-uma-shankar**
#### Changes proposed in this Pull Request

Implementation of the DIFM upsell offer A/B test. For more context, check pcbrnV-12W-p2.

A quick summary of changes:

* A/B test the DIFM upsell offer copy for users who purchase a Business or eCommerce plan. Check this [Figma](https://www.figma.com/file/b6jGfIqCpucVeNHlx82LA3/DIFM-post-purchase-offer-flow?node-id=106%3A9) for the design.
* Business plan users who decline the upsell will be taken to Customer Home. Those who Accept the offer, they will be taken to an intake form page which displays embedded the Crowdsignal DIFM survey. Successful completion of the survey will take the user to Customer Home.
* eCommerce users who decline the upsell will be taken to the Atomic Thank You page. Those who Accept the offer will be taken to an intake form page which displays embedded the Crowdsignal DIFM survey. Successful completion of the survey will take the user to Customer Home.

**DIFM Upsell offer**

Screenshot 2021-01-28 at 12 05 18 AM

**DIFM intake form**

Screenshot 2021-01-28 at 12 04 36 AM

#### Testing instructions

**Apply patch D56268-code**

**Scenario 1**

* Assign yourself to the treatment of the `post_purchase_difm_upsell` experiment.
* Go through the signup flow and purchase a Business plan. Verify that you see the DIFM upsell offer.
    - Declining the offer should take to thank you page.
    - Accepting the offer should take to a DIFM intake survey. Successful completion of the survey should take to Customer Home with a [notice indicating successful form submission](https://d.pr/i/ZsR2rv). Note that the form redirects to wordpress.com on successful submission, so replace the domain with calypso url to see the success notice.
    - You should receive an email with a link to the DIFM intake page on accepting the upsell offer. 
* Go through the signup flow and purchase an eCommerce plan. Verify that you see the DIFM upsell offer.
    - Declining the offer should take to the [atomic Thank You page](https://d.pr/i/BoQqLk)
    - Accepting the offer should take to a DIFM intake survey. Successful completion of the survey should take to [atomic Thank You page](https://d.pr/i/BoQqLk).
    - You should receive an email with a link to the DIFM intake page on accepting the upsell offer.
    
**Scenario 2**

* Assign yourself to `control` of the `post_purchase_difm_upsell` experiment.
* Verify that control experience is unchanged for Business purchase, After purchasing a Business plan, you're either taken to Customer Home or shown a Quick Start upsell, depending on which variant you get assigned on the Quick Start offer test.
* Verify that control experience is unchanged for eCommerce purchase. After purchasing an eCommerce plan, you should be taken to the atomic Thank You page.

### CODE REVIEW

File: client/my-sites/checkout/upsell-nudge/difm-upsell/index.jsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+							{ translate(
+								'There are more than 50,000 WordPress plugins available to help turn your site into any powerful platform you imagine.'
+							) }
+						</p>
+						<p>{ translate( 'Our premium website building service is perfect for:' ) }</p>
+						<ul className="difm-upsell__checklist">
+							<li className="difm-upsell__checklist-item">
+								<Gridicon icon="checkmark" className="difm-upsell__checklist-item-icon" />
+								<span className="difm-upsell__checklist-item-text">
+									{ translate( 'Online stores', {
```
Comment (a8ci18n) [inline client/my-sites/checkout/upsell-nudge/difm-upsell/index.jsx:101]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **30** times:
`translate( 'Best for Online Stores:' )` *ES Score: 8*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/difm-upsell-ab-test)

File: client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/get-thank-you-page-url.ts [hunk 2]
```diff
@@ -364,8 +370,22 @@ function getRedirectUrlForConciergeNudge( {
 				}
 			}
 
+			if ( hasBusinessPlan( cart ) ) {
+				if ( shouldShowDifmUpsell ) {
+					recordTracksEvent( 'calypso_eligible_difm_upsell' );
+					return `/checkout/${ siteSlug }/offer-difm/${ pendingOrReceiptId }`;
```
Comment (sirbrillig) [inline client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/get-thank-you-page-url.ts]: Can you please add tests to `client/my-sites/checkout/composite-checkout/test/composite-checkout-thank-you.js` to cover these new return values?

**Comment by victorespigares** [comment]
Checked from UX perspective, couple of details:

- Let's add an additional line in the header form: `We'll get back to you in 2-3 business days`. Check screenshot.

image

- ~While trying Business, I submitted the form and didn't get the success notification in Customer Home. The URL did have the `?notice=difm_success`though.~ Oh wait... replicated it and noticed I was being taking to wordpress.com, hence the lack of notification. Replacing the domain for the calypso.live link and the notification is there.
- Copy is an old one in the offer, but I'm sure you're aware of this :)
- I'm trying to test eCommerce, but haven't been able to. I'm getting an error that says "User or Token does not have access to specified site". Screenshot:

image

Things that are weird in the signup flow is that I'm not getting a .wordpress.com subdomain in the Domains step. Not sure where this is coming.

**Comment by niranjan-uma-shankar** [comment]
> Let's add an additional line in the header form: We'll get back to you in 2-3 business days.

Done 

> replicated it and noticed I was being taking to wordpress.com, hence the lack of notification. Replacing the domain for the calypso.live link and the notification is there.

Sorry, I missed to mention it. I've now updated this point in the testing instructions.

> Copy is an old one in the offer,

I've updated the PR with the latest copy. Also, I've implemented a new endpoint that will send an automated email when the user clicks on the "Interested" CTA in the offer page. 

> 'm trying to test eCommerce, but haven't been able to. I'm getting an error that says "User or Token does not have access to specified site". Things that are weird in the signup flow is that I'm not getting a .wordpress.com subdomain in the Domains step. Not sure where this is coming.:

It seems to be some permissions issue.

File: client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/get-thank-you-page-url.ts (hunk 2 above)
Comment (niranjan-uma-shankar) [reply client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/get-thank-you-page-url.ts]: Test added, thanks for the feedback.

File: client/my-sites/checkout/checkout-thank-you/index.jsx [hunk 3]
```diff
@@ -437,6 +446,14 @@ export class CheckoutThankYou extends React.Component {
 				);
 			}
 
+			if (
+				shouldShowDifmUpsell &&
+				! previousRoute.includes( `/checkout/${ selectedSiteSlug }/offer-difm/${ receiptId }` )
+			) {
+				recordTracksEvent( 'calypso_eligible_difm_upsell' );
```
Comment (niranjan-uma-shankar) [inline client/my-sites/checkout/checkout-thank-you/index.jsx]: Custom Tracks event is created to be used as an exposure event for the experiment. This helps give more control on which users we want to enter into the experiment.

File: client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/get-thank-you-page-url.ts [hunk 4]
```diff
@@ -364,8 +370,22 @@ function getRedirectUrlForConciergeNudge( {
 				}
 			}
 
+			if ( hasBusinessPlan( cart ) ) {
+				if ( shouldShowDifmUpsell ) {
+					recordTracksEvent( 'calypso_eligible_difm_upsell' );
```
Comment (niranjan-uma-shankar) [inline client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/get-thank-you-page-url.ts]: Same as above. Custom Tracks event is created to be used as an exposure event for the experiment. This helps give more control on which users we want to enter into the experiment.

**Review by niranjan-uma-shankar — COMMENTED** [review]
@Automattic/experimentation-platform I wanted to check with you all if this is okay. I needed to added a custom Tracks event for this experiment and I don't expect it to be necessary when the test wraps up(win or lose).

File: client/my-sites/checkout/checkout-thank-you/index.jsx (hunk 3 above)
Comment (jessie-ross) [reply client/my-sites/checkout/checkout-thank-you/index.jsx]: Regarding my comment, you might need to split this `if` into two.

**Review by jessie-ross — CHANGES_REQUESTED** [review]
> Custom Tracks Event

Hi @niranjan-uma-shankar!

It seems like you are only sending the event under treatment, it should also be sending the event in the case of control - i.e. moving it above `if ( shouldShowDifmUpsell ) {`. The reasoning is that Exposure events are for marking exposure to an experiment - not a variation.

See also:
https://github.com/Automattic/experimentation-platform/wiki/Experimenter's-Guide#what-is-an-exposure-event-and-when-do-i-need-it

**Comment by niranjan-uma-shankar** [comment]
> It seems like you are only sending the event under treatment, it should also be sending the event in the case of control

Thanks for spotting that @jessie-ross. I've pushed in a fix for it, could you please review again?

File: client/my-sites/checkout/checkout-thank-you/index.jsx [hunk 5]
```diff
@@ -438,6 +447,15 @@ export class CheckoutThankYou extends React.Component {
...
 			}
 
+			// This is for the DIFM upsell A/B test. Check https://wp.me/pcbrnV-Y3.
+			recordTracksEvent( 'calypso_eligible_difm_upsell' );
+			if (
+				shouldShowDifmUpsell &&
+				! previousRoute.includes( `/checkout/${ selectedSiteSlug }/offer-difm/${ receiptId }` )
+			) {
+				page( `/checkout/${ selectedSiteSlug }/offer-difm/${ receiptId }?isEcommerce=1` );
+			}
```
Comment (jessie-ross) [inline client/my-sites/checkout/checkout-thank-you/index.jsx]: I think it is more accurate to split this?
```suggestion
			// This is for the DIFM upsell A/B test. Check https://wp.me/pcbrnV-Y3.
			if ( ! previousRoute.includes( `/checkout/${ selectedSiteSlug }/offer-difm/${ receiptId }` ) ) {
				recordTracksEvent( 'calypso_eligible_difm_upsell' );

				if (  shouldShowDifmUpsell ) {
        		   	page( `/checkout/${ selectedSiteSlug }/offer-difm/${ receiptId }?isEcommerce=1` );
		    	}
			}
```

**Review by jessie-ross — CHANGES_REQUESTED** [review]
Just left a comment about splitting one of the `if` statements.

File: client/my-sites/checkout/checkout-thank-you/index.jsx (hunk 5 above)
Comment (niranjan-uma-shankar) [reply client/my-sites/checkout/checkout-thank-you/index.jsx]: I think it's okay to have the `recordTracksEvent()`  call before the `if` condition. The `previousRoute` check is only for those users who've already seen the offer page and we don't want to show it to them a second time. So, they've already entered the experiment at that point.

**Comment by taggon** [comment]
```
... users who purchase a Business or eCommerce plan
```
Why don't you create a custom segment filter to narrow the target audience? Please check `wpcom-experiment-custom-segments.php` file in D52254-code.

File: client/my-sites/checkout/checkout-thank-you/index.jsx (hunk 5 above)
Comment (jessie-ross) [reply client/my-sites/checkout/checkout-thank-you/index.jsx]: Yeah that makes sense to me :)

**Comment by niranjan-uma-shankar** [comment]
> Why don't you create a custom segment filter to narrow the target audience? Please check wpcom-experiment-custom-segments.php file in D52254-code.

@taggon Just so I can understand this better, could you please let me know the advantage in doing that? Looks like we would still need an exposure event of the type created in this PR `calypso_eligible_difm_upsell`, especially in the case of eCommerce since we only want those users who've successfully finished atomic transfer after eCommerce plan purchase to enter the experiment. It looks to me that exposure event is another way of controlling who can or cannot enter the experiment.

**Comment by taggon** [comment]
Upgraded an existing Premium site to Business and submitted the DIFM form then it redirects me to `/home?notice=difm_success` that shows the site selector instead of the success notification. Is this intended?

WordPress_com

**Comment by niranjan-uma-shankar** [comment]
> Upgraded an existing Premium site to Business and submitted the DIFM form then it redirects me to /home?notice=difm_success that shows the site selector instead of the success notification. Is this intended?

Good point. The reason this happens is because I'm embedding a Crowdsignal survey in the DIFM intake page and the success callback is defined as https://wordpress.com/home in Crowdsignal. The behaviour you're seeing is expected. It's probably sub-optimal that you were shown a site picker and it happens because you had more than one site. However, since this test is exposed to new users only, it's unlikely they'll see the site picker. Are you able to see the success notice when you select a site from the site picker? 


 I've also mentioned the point about wordpress.com redirect in the testing instructions:

> Note that the form redirects to wordpress.com on successful submission, so replace the domain with calypso url to see the success notice.

**Comment by niranjan-uma-shankar** [comment]
@taggon I pushed a change in [31a2685](https://github.com/Automattic/wp-calypso/pull/49301/commits/31a2685e8e6cbe536d4f79c0c3c2d92d4395d0d4) to reduxify the Customer Home success notice, keeping in line with the recent changes to reduxify all notices in https://github.com/Automattic/wp-calypso/issues/48408.

**Review by jessie-ross — APPROVED** [review]

**Comment by jessie-ross** [comment]
The ExPlat side of things LGTM :-)

**Review by victorespigares — APPROVED** [review]
LGTM from UX side! Tested eCommerce and was working fine. Also just added a commit with some small format changes for the offer.

**Review by taggon — APPROVED** [review]
> @taggon Just so I can understand this better, could you please let me know the advantage in doing that?

Never mind. I don't think it impacts much in this case. Now everything looks good to me.

File: client/my-sites/checkout/upsell-nudge/difm-upsell/index.jsx [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+										comment:
+											"This is a type of business listed on the 'Website building service' offer page",
+									} ) }
+								</span>
+							</li>
+						</ul>
+
+						<p>
+							{ translate(
+								'Simply tell us a little bit more about your website, and we’ll get back to you in 2-3 business days.'
```
Comment (a8ci18n) [inline client/my-sites/checkout/upsell-nudge/difm-upsell/index.jsx:138]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **1** times:
`translate( 'Fill out the application form and tell us a little bit more about your website. We’ll get back to you in 2-3 business days.' )` *ES Score: 6*

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/5501539

Thank you @niranjan-uma-shankar for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/5501539).

**Pull request merged**